packages/vue-app/template/router.js

```diff
@@ -100,6 +100,10 @@ export function routerOptions() {
   }
 }
 
-export function createRouter() {
-  return new Router(routerOptions())
+export async function createRouter(ssrContext, options = routerOptions) {
+  if (typeof options === 'function') {
+    options = await options(ssrContext)
```

Comment:
await is unnecessary and costly. we can set default to `options = routerOptions()` and when calling with custom arg, await before calling `createRouter` :)

Reply:
Can't pass `ssrContext`? I thought it might be useful for some kind of customization

Reply:
Hmm actually what we are doing is:
- passing ssrContext and routerOptions to createRouter
- Let create router call options with ssrContext (with extra await)
- Use it to create router instance

This can be simply:

- Using ssrContext to create options object (can be await or not)
- Pass it to createRouter

```js
createRouter(options = routerOptions)
```

These are identical. Just refactor on function to make it simpler and lighter.

Reply:
But this can generate a break change for those who use `ssrContext` in` createRoute`

Reply:
I'm not sure. At least __currently__ nuxt accepts no arguments for `createRouter`. Maybe I'm misunderstanding how are you going to use it for router-module. Can you please provide more details about the usage of (new) exported function?

Reply:
Today accepts yes, see https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/template/index.js#L59

Reply:
Today I do not have access to router options without initialization

Reply:
I just wanted to have access to router options without having to boot

Reply:
Is see. So isn't just exporting `routerOptions` functon enough to import from `defaultRouter`?

Reply:
Exactly, I just need to export routeOptions

Reply:
When we enable keepDefaultRouter
https://github.com/nuxt-community/router-module#accessing-default-router

The default router is created and I return a new router, ie the router is being started twice, I want to avoid this by using the default router options

Reply:
We can simplify by doing this:
```js
export const routerOptions = {
...
}

export async function createRouter(ssrContext) {
  return new Router(routerOptions)
}
```